### DISCUSSION TITLE
Is the truncation code in extract.py reasonable?

### DISCUSSION THREAD

**Discussion by diamondgloves** [facebookresearch/esm#156] [Q&A]
When I use ESM-1b model to extract representations of the sequences with lengths over 1024, the tokens of the sequence is supposed to be 'bos + seq + eos' with length of seq_len+2.According to the code in extract.py, toks becomes 'bos+seq[:1021]' without eos before feeding the model, could this be a reasonable input?
```
if args.truncate:
    toks = toks[:, :1022]
out = model(toks, repr_layers=repr_layers, return_contacts=return_contacts)
```
Furthermore, in ContactPredictionHead the contact map will be cropped into size of 1020*1020 with the aim at removing the bos&eos, but there is no eos in the toks.
Should the input toks of the model be as 'bos + seq[:1022] + eos' with length of 1024 after truncating?

**Comment by tomsercu** [comment]
Thanks for opening the issue #157, indeed the logic is incorrect.
